aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -631,6 +631,119 @@ void qclamp_kernel(
   });
 }
 
+void qclamp_with_tensors_kernel(
```

Comment:
I think you need to check several things: 1) Tensor shapes being the same, 2) Quantization parameters being the same.

For number (2) I am just thinking if the tensors are quantized using different scale and zero_point, that might lead to incorrect results

Reply:
thanks for working on this!  +1, we would need to handle quantization parameters being different between qx, min_tensor, max_tensor in the kernel.

Reply:
Thanks for reviewing! I've added tests to check for different quantization parameters. I did not add tests however for different shapes as that is handled by default by the TensorIterator handler (default broadcasting rules) and is not specific to quantization, so I believe it is not necessary.

Reply:
if the quantization parameters are the same between qx, min_tensor, max_tensor then this would work.  Would be good to also handle the case when quantization params are different.  We can handle it by dequantizing, clamping in floating point, and requantizing using the quantization params of `qx`.  In pseudocode,
 
`
qx_dq = qx.dequantize();
min_tensor_dq = min_tensor.dequantize();
max_tensor_dq = max_tensor.dequantize();
... clamp in floating point ...
qy = clamped.quantize_per_tensor(...qx's quantization params...);
return qy;
`

Reply:
Thanks! I've implemented it referencing the fp implementation now + tests for that.